src/tiled/toolspecifictoolbar.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * toolspecifictoolbar.h
+ * Copyright 2016, Ketan Gupta <ketan19972010@gmail.com>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <QIcon>
+#include <QToolBar>
+
+class QToolButton;
+
+namespace Tiled {
+namespace Internal {
+
+class AbstractTool;
+class MapEditor;
+
+class ToolSpecificToolBar : public QToolBar
+{
+public:
+    ToolSpecificToolBar(QWidget *parent = nullptr, MapEditor *mapEditor = nullptr);
```

Comment:
It would do that if you forget the first part of my initial comment, which was to make the parent parameter the last one. In C++, once some parameter is given a default value, all parameters after that also need a default value.